### PR TITLE
Run CDN in a Box for Developers services as unprivileged users

### PR DESCRIPTION
**Pull request by zrhoffman**
This PR makes it so that when the CDN in a Box for Developers services create new files (from `npm ci`, building debug binaries, creating Traffic Router DBs, etc.), the files are owned by the user who owns the repository directory, rather than being owned by root.
<hr/>

## Which Traffic Control components are affected by this PR?

- CDN in a Box - for Developers

## What is the best way to verify this PR?

1. Remove generated files and directories:
   ```shell
   rm -rf dev/traffic_router/db experimental/traffic-portal/node_modules traffic_monitor/__debug_bin traffic_ops/traffic_ops_golang/__debug_bin traffic_portal/node_modules
   ```
2. Build the CDN in a Box for Developers:
   ```shell
   docker-compose build
   ```

3. Start the Dev CiaB:
   ```shell
   docker-compose up
   ```
4. Verify that the generated files and directories are owned by you
```shell
ls -ld dev/traffic_router/db experimental/traffic-portal/node_modules traffic_monitor/__debug_bin traffic_ops/traffic_ops_golang/__debug_bin traffic_portal/node_modules
```

## PR submission checklist
- [ ] This PR has tests 
- [ ] This PR has documentation 
- [ ] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Comment by ocket8888** [comment]
This is something I've wanted for the dev environment since its inception, but I can't test it because of a bug in the TP/TPv2 containers that prevents them from fetching a resource from npm. I've had another developer confirm that the issue is not specific to my machine, but I still have no idea why that's happening.

**Comment by zrhoffman** [comment]
Rebased onto master to get #7164

File: dev/tpv2/run.sh [hunk 1]
```diff
@@ -16,5 +16,24 @@
 set -o errexit
 
 cd "$TC/experimental/traffic-portal"
+
+user=tpv2
+uid="$(stat -c%u "$TC")"
+gid="$(stat -c%g "$TC")"
+if [[ "$(id -u)" != "$uid" ]]; then
+	for dir in "${TC}/.npm" .angular node_modules; do
```
Comment (ocket8888) [inline dev/tpv2/run.sh:24]: this fails on fresh repo clones:
```
tpv2_1            | stat: cannot statx '.angular': No such file or directory
tpv2_1            | chown: cannot access '.angular': No such file or directory
```
(and then the container crashes)

**Review by ocket8888 — CHANGES_REQUESTED** [review]

File: dev/tpv2/run.sh (hunk 1 above)
Comment (zrhoffman) [reply dev/tpv2/run.sh:24]: Checking if the directory exists first in ec0651f0e2

**Review by ocket8888 — COMMENTED** [review]
Most things appear to be working, but it seems that the Go binaries it generates are still owned by root:
```console
$ ls -lR . | grep -v "$(whoami)" | grep root | grep -vE '^\..+:$'  
-rwxr-xr-x 1 root      root      6881051 Nov  9 10:57 t3c
-rwxr-xr-x 1 root      root      9965720 Nov  9 10:57 t3c-apply
-rwxr-xr-x 1 root      root      6876971 Nov  9 10:57 t3c-check
-rwxr-xr-x 1 root      root      6958591 Nov  9 10:57 t3c-check-refs
-rwxr-xr-x 1 root      root      6854750 Nov  9 10:57 t3c-check-reload
-rwxr-xr-x 1 root      root      6958920 Nov  9 10:57 t3c-diff
-rwxr-xr-x 1 root      root      8375868 Nov  9 10:57 t3c-generate
-rwxr-xr-x 1 root      root      6960525 Nov  9 10:57 t3c-preprocess
-rwxr-xr-x 1 root      root      10250226 Nov  9 10:57 t3c-request
-rwxr-xr-x 1 root      root      7076256 Nov  9 10:57 t3c-tail
-rwxr-xr-x 1 root      root      9794976 Nov  9 10:57 t3c-update
-rwxr-xr-x 1 root      root      11172798 Nov  9 10:57 tc-health-client
-rwxr-xr-x 1 root      root      7106164 Nov  9 10:56 admin
```

File: docker-compose.yml [hunk 2]
```diff
@@ -33,8 +33,10 @@ services:
         ports:
             - 6443:443
             - 6444:6444
+        sysctls:
+            - net.ipv4.ip_unprivileged_port_start=0
```
Comment (ericholguin) [inline docker-compose.yml:37]: Maybe should be changed to 23 so ssh port isnt exposed?

Comment (zrhoffman) [reply docker-compose.yml:37]: This isn't which ports are exposed, it's which ports a non-user is allowed to listen on.

Docker itself [includes](https://github.com/moby/moby/pull/41030/files#diff-9f91bff23e0bd70d6429b63d9db2d8180d2e89cdb64db4fb3e10a96f74d36271R791-R793) `net.ipv4.ip_unprivileged_port_start = 0`, just not containerd yet.

**Comment by zrhoffman** [comment]
Oops, the condition checking if t3c's `run.sh` was being run as root was always checking the unprivileged user's UID instead. Fixed in 0150a22289

**Review by ocket8888 — APPROVED** [review]

**Pull request merged**